src/main/java/ru/hogwarts/school/service/StudentService.java

```diff
@@ -0,0 +1,57 @@
+package ru.hogwarts.school.service;
+
+import org.springframework.stereotype.Service;
+import ru.hogwarts.school.model.Faculty;
+import ru.hogwarts.school.model.Student;
+import ru.hogwarts.school.repositories.StudentRepository;
+
+import java.util.Collection;
+import java.util.stream.Collectors;
+
+@Service
+public class StudentService {
+    private final StudentRepository studentRepository;
+
+    public StudentService(StudentRepository studentRepository) {
+        this.studentRepository = studentRepository;
+    }
+
+    public Student createStudent(Student student) {
+        return studentRepository.save(student);
+    }
+
+    public Student readStudent(Long id) {
+        if (studentRepository.existsById(id)) return studentRepository.findById(id).get();
```

Comment:
Это всё делается одной строкой, если правильно использовать Optional:
```suggestion
        return studentRepository.findById(id).orElse(null);
```